indexers/vector/SptagIndexer/__init__.py

```diff
@@ -44,11 +44,15 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
         _index.SetBuildParam("NumberOfThreads", str(self.num_threads))
         _index.SetBuildParam("DistCalcMethod", self.space)
 
-        if _index.Build(vecs, vecs.shape[0]):
+        if _index.Build(vecs.astype('float32'), vecs.shape[0]):
```

Comment:
In their example, they Build, save to a new directory and load from the directory? Is it needed here? And also they have a different mode where they use a client to a remote indexer

Reply:
I'm following this tutorial, https://github.com/microsoft/SPTAG/blob/master/docs/Tutorial.ipynb

Reply:
Yes, this is the one I am talking about, but maybe the `Save` is just for persistence and not necessary. If the tests pass it is good

Reply:
Yes, `Save` is  for saving the index for later usages. It is not required by the `Search`.